todo-app/src/components/ListItem.js

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import PropTypes from "prop-types";
+import UpdateTodo from "./UpdateTodo";
+
+
+const ListItem = (({ todoItems, todo }) => {
+
+    const {
+        toggleCheck,
+        toggleEdit,
+        deleteTodo
+    } = todoItems;
+
+    const {
+        id,
+        description,
+        deadline,
+        done,
+        editMode
+    } = todo;
+
+    const date = deadline.split("-").join("/");
+    const crossOut = done ? "crossOut" : null;
+
+    return (
+        <section>
+            {
+                editMode
+                ?
+                <UpdateTodo todo={todo}
```

Comment:
You're passing redundant data here, no? You can either pass the individual todo object and the update function, or the entire store and just the todo's ID.